### DISCUSSION TITLE
ReLU function in casadi

### DISCUSSION THREAD

**Discussion by imanask** [casadi/casadi#3678] [Q&A]
I am trying to do MPC with a neural network as the dynamic model with a ReLU activation Layer. There is no default ReLU function by CasADi and MATLAB's relu function and reluLayer. As a last resort, I used if_else(x> 0, x, 0); This makes the problem very difficult to solve. Is there a way to deal with the ReLU function in CasADi?

Side note: I also tried a network with Tanh activation. In this case, I used MATLAB's tanh function which apparently work fine with MX variable.

**Comment by sandeep026** [comment]
Tried softplus function? It will the make the problem more nonlinear though

**Reply by imanask** [reply]
Thanks for the suggestion. Yes, this is what I ended up doing. It appears that ReLU functions are difficult to deal with IPOPT. I found that the literature also mentions the same (https://arxiv.org/pdf/2111.10489, Section 3.1 and Fig 5).

**Reply by sandeep026** [reply]
Glad that it helped
